admin_guide/backup_restore.adoc

```diff
@@ -0,0 +1,263 @@
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+
+// REVIEWERS: READ THIS!
+//
+// In the following text, there are questions of the form:
+//   //??? QUESTION
+// Please feel free to make a line-comment to answer them, in addition
+// to any other (line-)comments on the correctness of the text.
+//
+// - Usually, the question pertains to the text preceding it.
+//   Questions pertaining to following text are explicitly noted.
+//
+// - There are bunch of questions at the end.
+//
+// Thanks for your cooperation on this (experimental) method of
+// refining the documentation.  Hopefully it will bear good fruit.
+
+== Overview
+
+In {product-title}, you can
+_back up_ (saving state to separate storage)
+and _restore_ (recreating state from separate storage)
+at the cluster level.
+There is also some preliminary support for
+xref:project-backup[per-project backup].
+The full state of a cluster installation includes:
+
+- etcd data on each master
+- API objects
+- registry storage
+- volume storage
+
+This topic does not cover how to back up and restore
+link:../install_config/persistent_storage/index.html[persistent storage],
+as those topics are left to the underlying storage provider.
+
+
+[[backup-restore-prerequisites]]
+== Prerequisites
+
+Because the restore procedure involves a
+link:#cluster-restore[complete reinstallation],
+we recommend that you save all the files used in the initial installation.
+This may include:
+
+- *_~/.config/openshift/installer.cfg.yml_* (from the
+link:../install_config/install/quick_install.html[Quick Installation]
+method)
+- ansible playbooks and inventory files (from the
+link:../install_config/install/advanced_install.html[Advanced Installation]
+method)
+- *_/etc/yum.repos.d/ose.repo_* (from the
+link:../install_config/install/disconnected_install.html[Disconnected Installation]
+method)
+//??? Other files?
+
+Install packages that provide various utility commands:
+
+----
+# yum install etcd
+----
+
+Note the location of the *etcd* data directory
+(or `$ETCD_DATA_DIR` in the following sections),
+which depends on how *etcd* is deployed.
+
+[options="header",cols="1,2"]
+|===
+| Deployment | Data Directory
+
+|all-in-one VM
+|*_/var/lib/openshift/openshift.local.etcd_*
+
+|external (not on master)
+|*_/var/lib/etcd_*
+
+|embedded (on master)
+|*_/var/lib/origin/etcd_*
+|===
+
+
+[[cluster-backup]]
+== Cluster Backup
+
+. Save all the certificates and keys, on each master:
++
+----
+# cd /etc/origin/master
+# tar cf /tmp/certs-and-keys-$(hostname).tar \
+    master.proxy-client.crt \
+    master.proxy-client.key \
+    proxyca.crt \
+    proxyca.key \
+    master.server.crt \
+    master.server.key \
+    ca.crt \
+    ca.key \
+    master.etcd-client.crt \
+    master.etcd-client.key \
+    master.etcd-ca.crt
+----
+//??? What is missing?
+//??? What is unnecessary?
+
+. Create an *etcd* backup:
++
+----
+# etcdctl backup \
+    --data-dir $ETCD_DATA_DIR \
+    --backup-dir $ETCD_DATA_DIR.bak
+----
++
+[NOTE]
+====
```

Comment:
Quick feedback. One of the users asked us if this etcd backup can be done without stopping service or not. I feel that it will be FAQ, so could you please add it to the doc?

NOTE: Although I thought `etcdctl backup` can do live backup, etcd's documentation stopped service.
https://coreos.com/etcd/docs/latest/etcd-live-cluster-reconfiguration.html#etcd-disaster-recovery-on-coreos

```
Let’s assume a 3-node cluster with no living members. First, stop the etcd2 service on all the members:

$ sudo systemctl stop etcd2
```